[Spark-14300][Docs][MLLIB]Scala MLlib examples code merge and clean up

keypointt opened the pull request:
## What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/SPARK-14300

Duplicated code found in scala/examples/mllib, below all deleted in this PR:
- DenseGaussianMixture.scala
- StreamingLinearRegression.scala
## delete reasons:
#### delete: mllib/DenseGaussianMixture.scala
- duplicate of mllib/GaussianMixtureExample
#### delete: mllib/StreamingLinearRegression.scala
- duplicate of mllib/StreamingLinearRegressionExample

When merging and cleaning those code, be sure not disturb the previous example on and off blocks.
## How was this patch tested?

Test with `SKIP_API=1 jekyll` manually to make sure that works well.

SparkQA commented:
**[Test build #55083 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55083/consoleFull)** for PR 12195 at commit [`b469816`](https://github.com/apache/spark/commit/b4698164cabdb76e019aed5e5f55976c547e606c).

keypointt commented:
#### Unsure code duplications (need doube check)

- AbstractParams.scala - cannot delete this one, not duplicate, its usage found in many examples

- BinaryClassification.scala - @pending: not sure, duplicate of mllib/BinaryClassificationMetricsExample ???
    
- CosineSimilarity.scala - @pending: personally I prefer to keep it as a demo
    
- MovieLensALS.scala - @pending: duplicate of LocalALS.scala, but I prefer to delete LocalALS.scala but keep MovieLensALS.scala, since MovieLensALS better commented
   
- MultivariateSummarizer.scala - @pending: usage not found, but better keep it for possible later demo
    
- RandomRDDGeneration.scala - @pending: usage not found, but better keep it for possible later demo
    
- SampledRDDs.scala - @pending: usage not found, but better keep it for possible later demo

hi @yinxusen, for above @pending ones, I'm not sure we should delete them or not? There are no usage found for them, but personally I find they are still useful examples

SparkQA commented:
**[Test build #55083 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55083/consoleFull)** for PR 12195 at commit [`b469816`](https://github.com/apache/spark/commit/b4698164cabdb76e019aed5e5f55976c547e606c).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55083/
Test PASSed.

yinxusen commented:
@keypointt As @mengxr said here https://github.com/apache/spark/pull/12092#issuecomment-204276885, we should keep examples such as `DecisionTreeRunner`.

yinxusen commented:
So I think we should keep

* mllib/DecisionTreeRunner.scala
* mllib/GradientBoostedTreesRunner.scala
* mllib/DenseKMeans.scala
* mllib/LDAExample.scala
* mllib/LinearRegression.scala
* mllib/SparseNaiveBayes.scala

yinxusen commented:
And for examples in **Unsure code duplications (need doube check)** you mentioned above, I think we should keep them, too.

SparkQA commented:
**[Test build #55873 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55873/consoleFull)** for PR 12195 at commit [`7962626`](https://github.com/apache/spark/commit/7962626b0057cb1830c5502ce38441540741de35).

SparkQA commented:
**[Test build #55873 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55873/consoleFull)** for PR 12195 at commit [`7962626`](https://github.com/apache/spark/commit/7962626b0057cb1830c5502ce38441540741de35).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55873/
Test FAILed.

keypointt committed:
[SPARK-14300] delete duplicates in scala mllib

keypointt commented:
@yinxusen I've added back the files you mention above, and also updated the description (top comment) in this PR

SparkQA commented:
**[Test build #55876 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55876/consoleFull)** for PR 12195 at commit [`b6a63dd`](https://github.com/apache/spark/commit/b6a63ddf0eb277cc5597e693b5ae7916658bd463).

SparkQA commented:
**[Test build #55876 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/55876/consoleFull)** for PR 12195 at commit [`b6a63dd`](https://github.com/apache/spark/commit/b6a63ddf0eb277cc5597e693b5ae7916658bd463).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/55876/
Test PASSed.

keypointt commented:
@yinxusen I've added back the files you mention above, and also updated the description (top comment) in this PR. In this PR I only deleted couple files, is there anything extra I should do?

yinxusen commented:
@keypointt Sorry for the late. I'm  taking a look now.

yinxusen commented:
@mengxr LGTM

keypointt commented:
@yinxusen No problem, I totally understand you guys are super busy 😄

keypointt commented:
ping @mengxr

SparkQA commented:
**[Test build #65450 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65450/consoleFull)** for PR 12195 at commit [`b6a63dd`](https://github.com/apache/spark/commit/b6a63ddf0eb277cc5597e693b5ae7916658bd463).

SparkQA commented:
**[Test build #65450 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65450/consoleFull)** for PR 12195 at commit [`b6a63dd`](https://github.com/apache/spark/commit/b6a63ddf0eb277cc5597e693b5ae7916658bd463).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/65450/
Test PASSed.

jkbradley commented on examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegression.scala:
```diff
@@ -1,73 +0,0 @@
-import org.apache.spark.streaming.{Seconds, StreamingContext}
-
-/**
- * Train a linear regression model on one stream of data and make predictions
```
It looks like this documentation was not copied to the replacement ```StreamingLinearRegressionExample```.  Could you please copy it there?

jkbradley commented on examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLogisticRegression.scala:
```diff
@@ -1,75 +0,0 @@
- * Anytime you add text files to `testDir`, you'll see predictions from the current model.
- *
- */
-object StreamingLogisticRegression {
```
I don't see this example anywhere else.  Am I missing something?  If not, then let's leave it.

jkbradley commented on examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnyPCA.scala:
```diff
@@ -1,62 +0,0 @@
- * }}}
- * represents a 3-by-2 matrix, whose first row is (0.5, 1.0).
- */
-object TallSkinnyPCA {
```
Same here.  What is this duplicating?

jkbradley commented on examples/src/main/scala/org/apache/spark/examples/mllib/TallSkinnySVD.scala:
```diff
@@ -1,62 +0,0 @@
- * }}}
- * represents a 3-by-2 matrix, whose first row is (0.5, 1.0).
- */
-object TallSkinnySVD {
```
Ditto

keypointt replied:
sure Joseph, I'm adding these files back

jkbradley reviewed (commented):
Sorry to leave your PR hanging, but if you have time now, it'd be good to fix up and merge.  Thanks!

keypointt committed:
Merge branch 'master' into SPARK-14300

keypointt commented:
The main reason deleting those without duplication is because of the name, but the actual code differs:

* StreamingLinearRegression
https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegression.scala

* StreamingLinearRegressionExample
https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/mllib/StreamingLinearRegressionExample.scala

keypointt committed:
[SPARK-14300] add back files which are not duplications

SparkQA commented:
**[Test build #67486 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/67486/consoleFull)** for PR 12195 at commit [`2a23852`](https://github.com/apache/spark/commit/2a238520cdd05eba4ac8e92a58117140fb8063c5).

SparkQA commented:
**[Test build #67486 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/67486/consoleFull)** for PR 12195 at commit [`2a23852`](https://github.com/apache/spark/commit/2a238520cdd05eba4ac8e92a58117140fb8063c5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/67486/
Test PASSed.

keypointt commented:
I'll update the PR description if it looks good now

jkbradley commented:
@keypointt I think it's OK to delete StreamingLinearRegression.scala, as you were doing before your last commit.  For that one, my only complaint was that the similar example (StreamingLinearRegressionExample.scala) did not include the Scala doc describing the input data.  If you copy that doc into StreamingLinearRegressionExample.scala, then it will be OK to remove StreamingLinearRegression.scala.

keypointt committed:
[SPARK-14300] remove StreamingLinearRegression.scala, add description to StreamingLinearRegressionExample.scala

SparkQA commented:
**[Test build #67554 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/67554/consoleFull)** for PR 12195 at commit [`7bb5d9f`](https://github.com/apache/spark/commit/7bb5d9f5ab40e03e7e01bf44199d1860628138c9).

keypointt commented:
I've also updated the description of this PR

SparkQA commented:
**[Test build #67554 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/67554/consoleFull)** for PR 12195 at commit [`7bb5d9f`](https://github.com/apache/spark/commit/7bb5d9f5ab40e03e7e01bf44199d1860628138c9).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/67554/
Test PASSed.

jkbradley commented:
LGTM
merging with master
Thanks @keypointt !

keypointt commented:
thank you @jkbradley for the detailed explanation :)
